[docs] Add expo optimize description and usage info

sergeichestakov opened the pull request:
# Why

- Update output of `expo --help` in docs to show recent changes including `optimize` and how to learn more about a specific command. 
- Add more information about `expo optimize` in general in both the `assets` guide and `publishing` workflow pages.

Commit:
[docs] Update expo-cli output

Commit:
Update sdk 32 docs as well

Commit:
Add information about the optimize command

sergeichestakov commented:
I'll also add a note about the `--quality` flag if https://github.com/expo/expo-cli/pull/577 is approved and merged.

sergeichestakov commented:
Side note: The `android_test_suite` workflow has been finicky and occasionally failing for the past couple PRs (this one included). Need to look into if they're all for the same reason but it seems as though a test calling `scanFromURLAsync` is failing in the `BarCodeScanner` module.

esamelson reviewed (commented):
looks good, just one suggestion

esamelson commented on docs/pages/versions/unversioned/guides/assets.md:
```diff
@@ -20,4 +20,10 @@ Each time you publish your app, Expo will upload your assets to Amazon CloudFron
+
+### Images
+
+Images often take up the most space out of the assets in an Expo project. To compress the images (PNGs and JPGs) in your project, you can run `expo optimize`. Alternatively, when you publish using `expo publish`, you will be prompted and asked if you'd like to do so if you haven't already. You can pass in a `--save` flag to backup a copy of each file using a `.orig` extension so that you can easi ...
```
I would add a sentence or two here about why you want to do this (even if it seems obvious) since this is under the "Performance" header -- e.g. something like "optimizing your images will make them take up less space on users' devices, and will reduce the time and bandwidth needed to download before they are ready to display"

sergeichestakov replied:
Good call. Added in e46c3c5.

Left a note about the `--quality` flag as well.

Commit:
Add a sentence about why a user would want to optimize

Commit:
Add a description about the --quality flag

esamelson commented on docs/pages/versions/unversioned/guides/assets.md:
```diff
@@ -20,4 +20,10 @@ Each time you publish your app, Expo will upload your assets to Amazon CloudFron
+
+### Images
+
+Images often take up the most space out of the assets in an Expo project. Optimizing your images will make them take up less space on end users' devices and reduce the time and bandwidth needed to download before they are ready to display. To compress the images (PNGs and JPGs) in your project, you can run `expo optimize`. Alternatively, when you publish using `expo publish`, you will be prompted ...
```
Sorry to keep nitpicking here, but I still think this could be improved. It's a pretty big block of text right now and people generally have an easier time parsing text in smaller chunks (and are more likely to actually read it!). I'd suggest breaking this into 2 or 3 paragraphs, and/or putting all of the flags in a bulleted list rather than in prose.

sergeichestakov replied:
No worries, I appreciate the scrutiny. I want to make sure this looks good too so feel free to keep doing so. I addressed these concerns in 739470b and it's looking much better. Let me know if I need to change anything else.

Commit:
Rephrase image optimization section

- "Performance" -> "Optimization" (makes it easier to search for)
- Move "Images" above "Fonts"
- Switch to a brief description followed by a bulleted list of options

Commit:
JPG -> JPEG

esamelson reviewed (approved):
this is looking great @sergeichestakov ! 👍 go ahead and merge whenever you're ready.

Commit:
JPG -> JPEG in publishing section
